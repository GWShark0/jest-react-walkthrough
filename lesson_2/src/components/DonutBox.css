.donut-box {
  position: relative;
  display: inline-block;
  padding: 1rem 1rem 2rem;
  width: 480px;
  height: 360px;
  background-color: #B59176;
}

.donut-box::before {
  content: '';
  position: absolute;
  top: 3rem;
  right: 0.5rem;
  bottom: 0;
  left: 0.5rem;
  background-color: #997B65;
}

.donut-box::after {
  content: '';
  position: absolute;
  height: 3rem;
  right: 0;
  bottom: 0;
  left: 0;
  background-color: #FDB5CD;
}
