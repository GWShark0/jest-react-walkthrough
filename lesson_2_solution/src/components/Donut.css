.donut {
  position: relative;
  cursor: pointer;
  transition: transform 200ms ease-in-out;
}

.donut:hover {
  transform: translateY(-10%);
}
