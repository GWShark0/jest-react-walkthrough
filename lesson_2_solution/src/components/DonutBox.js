import React from 'react';
import Donut from './Donut';

import './DonutBox.css';

export const NUM_DONUTS = 12;

function DonutBox() {
  const donuts = Array.apply(null, Array(NUM_DONUTS));
  return (
    <div className="donut-box">
      {donuts.map((_, i) => <Donut key={i} />)}
    </div>
  );
}

export default DonutBox;
