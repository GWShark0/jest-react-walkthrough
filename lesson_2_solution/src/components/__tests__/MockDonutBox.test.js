import React from 'react';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';

import DonutBox, { NUM_DONUTS } from '../DonutBox';
import Donut from '../Donut';

jest.mock('../Donut', () => 'span');

describe('DonutBox', () => {

  it('should render (mocked)', () => {
    const component = renderer.create(<DonutBox />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

});
