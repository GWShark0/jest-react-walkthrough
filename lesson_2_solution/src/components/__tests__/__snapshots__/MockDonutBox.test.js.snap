// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DonutBox should render (mocked) 1`] = `
<div
  className="donut-box"
>
  <span />
  <span />
  <span />
  <span />
  <span />
  <span />
  <span />
  <span />
  <span />
  <span />
  <span />
  <span />
</div>
`;
